src/nvim/main.c

```diff
@@ -1972,6 +1972,41 @@ static bool do_user_initialization(void)
   return do_exrc;
 }
 
+// Read initialization commands from ".nvimrc.lua", ".nvimrc" or ".exrc" in
+// current directory.  This is only done if the 'exrc' option is set.
+// Only do this if VIMRC_FILE is not the same as vimrc file sourced in
+// do_user_initialization.
+static void do_exrc_initialization(void)
+{
+  char *str;
+
+  size_t len = strlen(VIMRC_FILE) + 5;
+  char *vimrc_lua = xmalloc(len);
+  xstrlcpy(vimrc_lua, VIMRC_FILE, len);
+  xstrlcat(vimrc_lua, ".lua", len);
+
+  if (os_path_exists(vimrc_lua)) {
+    str = nlua_read_secure(vimrc_lua);
+    if (str != NULL) {
+      do_source(vimrc_lua, false, DOSO_NONE);
```

Comment:
We cannot read the file with `vim.secure.read()` and then read it again off the file system. The point of `vim.secure.read()` and `nlua_read_secure()` is that the file contents are read into memory and _that_ is what is validated by the user. If the on-disk file contents changed in between, you've just been pwnd.

Use `nlua_exec` to execute a string of Lua code instead.